Corona-Warn-App/src/test/java/de/rki/coronawarnapp/eventregistration/checkins/split/CheckInSplitterTest.kt

```diff
@@ -0,0 +1,247 @@
+package de.rki.coronawarnapp.eventregistration.checkins.split
+
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import io.kotest.matchers.shouldBe
+import org.joda.time.Instant
+import org.junit.jupiter.api.Test
+
+import testhelpers.BaseTest
+
+/**
+ * Test examples reference [https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration-mvp/
+ * docs/spec/event-registration-client.md#split-checkin-by-midnight-utc]
+ */
+class CheckInSplitterTest : BaseTest() {
+
+    private val defaultCheckIn = CheckIn(
+        id = 1L,
+        guid = "eventOne",
+        version = 1,
+        type = 1,
+        description = "Restaurant",
+        address = "Around the corner",
+        traceLocationStart = null,
+        traceLocationEnd = null,
+        defaultCheckInLengthInMinutes = null,
+        signature = "signature",
+        checkInStart = Instant.now(),
+        checkInEnd = Instant.now(),
+        targetCheckInEnd = null,
+        createJournalEntry = false
+    )
+
+    @Test
+    fun `Scenario 1`() {
+        /*
+           Example 1 - same-day check-in
+           localCheckIn = { start: '2021-03-04 09:30', end: '2021-03-04 09:45' }
+           splitInto = [{ start: '2021-03-04 09:30', end: '2021-03-04 09:45' } // no split
+        */
+        val checkIn = defaultCheckIn.copy(
+            checkInStart = Instant.parse("2021-03-04T09:30:00Z"),
+            checkInEnd = Instant.parse("2021-03-04T09:45:00Z")
+        )
+
+        checkIn.splitByMidnightUTC().apply {
+            size shouldBe 1 // No splitting
+            get(0) shouldBe checkIn
+        }
+    }
+
+    @Test
+    fun `Scenario 2`() {
+        /*
+           Example 2 - no-checkout-time check-in
+           localCheckIn = { start: '2021-03-04 09:30', end: null }
+           splitInto = [{ start: '2021-03-04 09:30', end: null } // no split
+        */
+        val checkIn = defaultCheckIn.copy(
+            checkInStart = Instant.parse("2021-03-04T09:30:00Z"),
+            checkInEnd = null
+        )
+
+        checkIn.splitByMidnightUTC().apply {
+            size shouldBe 1 // No splitting
+            get(0) shouldBe checkIn
+        }
+    }
+
+    @Test
+    fun `Scenario 3`() {
+        /*
+           Example 3 - same-start-end-times check-in
+           localCheckIn = { start: '2021-03-04 09:30', end: '2021-03-04 09:30' }
+           splitInto = [{ start: '2021-03-04 09:30', end: '2021-03-04 09:30' } // no split
+        */
+        val checkIn = defaultCheckIn.copy(
+            checkInStart = Instant.parse("2021-03-04T09:30:00Z"),
+            checkInEnd = Instant.parse("2021-03-04T09:30:00Z")
+        )
+
+        checkIn.splitByMidnightUTC().apply {
+            size shouldBe 1 // No splitting
+            get(0) shouldBe checkIn
+        }
+    }
+
+    @Test
+    fun `Scenario 4`() {
+        /*
```

Comment:
Can you add a test for this scenario:      val checkIn = defaultCheckIn.copy(
            checkInStart = Instant.parse("2021-03-04T09:30:00Z"),
            checkInEnd = Instant.parse("2021-03-05T09:15:00Z")
        )
That fails for me. What happens when startTime < endTime?

Reply:
what do you mean by `startTime < endTime` ? is not that case in all test cases?

Reply:
Yes, I mean startTime > endTime, we should handle that possibility correctly.

Please test the example above, where only the time of the day is reversed, not the full date.

Reply:
I expect that there is no splitting should happen since the duration still less than one day, but I will align with Max regarding those cases? my understanding that the check-ins are split across multiple duration days. It does mean that they have to split for different dates

Reply:
@mlenkeit Can you confirm this?

Reply:
@chiljamgossow  I already aligned with Max regarding that. pushing a fix soon and a test cases for that